ports/espressif/common-hal/microcontroller/Pin.c

```diff
@@ -244,7 +249,7 @@ void reset_all_pins(void) {
         }
         _reset_pin(i);
     }
-    _in_use_pin_mask = _never_reset_pin_mask;
+    _in_use_pin_mask = _never_reset_pin_mask | pin_mask_reset_forbidden;
```

Comment:
so is this the "real fix", so that e.g., GPIO 33 is marked as in-use for CONFIG_SPIRAM_MODE_OCT? and the rest is [good] churn?  If so, good to go.

Reply:
Yup, that's the fix. The rest is meant to unify everything to using forbidden pins.